drivers/i2c/i2c_ll_stm32.c

```diff
@@ -90,13 +90,15 @@ static int i2c_stm32_init(struct device *dev)
 {
 	struct device *clock = device_get_binding(STM32_CLOCK_CONTROL_NAME);
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
-	struct i2c_stm32_data *data = DEV_DATA(dev);
+	u32_t bitrate_cfg;
 	int ret;
 
 	__ASSERT_NO_MSG(clock);
 	clock_control_on(clock, (clock_control_subsys_t *) &cfg->pclken);
 
-	ret = i2c_stm32_runtime_configure(dev, data->dev_config.raw);
+	bitrate_cfg = __i2c_map_dt_bitrate(cfg->bitrate) << 1;
```

Comment:
We should probably add 
```
#define I2C_SPEED_SHIFT 1
```
to include/i2c.h file and have here
```
bitrate_cfg = __i2c_map_dt_bitrate(cfg->bitrate) << I2C_SPEED_SHIFT;
```
or even better __i2c_map_dt_bitrate() could do the shift.